src/core/transaction.js

```diff
@@ -1413,6 +1501,59 @@ Transaction.prototype.summary = function() {
   return txSummary;
 };
 
-exports.Transaction = Transaction;
+Transaction.prototype.setSigners = function(signers) {
+  if (_.isArray(signers)) {
+    this.tx_json.Signers = signers;
+  }
+
+  return this;
+};
 
-// vim:sw=2:sts=2:ts=8:et
+Transaction.prototype.addMultiSigner = function(signer) {
+  assert(UInt160.is_valid(signer.Account), 'Signer must have a valid Account');
+
+  if (_.isUndefined(this.multi_signers)) {
+    this.multi_signers = [];
+  }
+
+  this.multi_signers.push({Signer: signer});
+
+  this.multi_signers.sort((a, b) => {
+    return UInt160.from_json(a.Signer.Account)
+    .cmp(UInt160.from_json(b.Signer.Account));
+  });
+};
+
+Transaction.prototype.hasMultiSigners = function() {
+  return !_.isEmpty(this.multi_signers);
+};
+Transaction.prototype.getMultiSigners = function() {
+  return this.multi_signers;
+};
+
+Transaction.prototype.getMultiSigningJson = function() {
+  assert(this.tx_json.Sequence, 'Sequence must be set before multi-signing');
+  assert(this.tx_json.Fee, 'Fee must be set before multi-signing');
+
+  const signingTx = Transaction.from_json(this.tx_json);
+  signingTx.remote = this.remote;
+  signingTx.setSigningPubKey('');
+  signingTx.setCanonicalFlag();
+
+  return signingTx.tx_json;
+};
+
+Transaction.prototype.multiSign = function(account, secret) {
+  const signingData = this.multiSigningData(account);
```

Comment:
We could make secret the first arg, and make account optional, using the seed -> keyPair.accountID()